_posts/2017-03-05-artsy-technology-stack-2017.md

```diff
@@ -0,0 +1,100 @@
+---
+layout: post
+title: Artsy's Technology Stack, 2017
+date: 2017-03-05
+categories: [Technology, eigen, force, gravity]
+author: orta
+series: Artsy Tech Stack
+---
+
+Artsy has now grown past 160 team members and our Engineering organization is now 30? strong. [WIP] For a brief overview of what the company has accomplished in the last few years, check out our [2013](http://2013.artsy.net) and [2014](http://2014.artsy.net) reviews.
+
+I'd like to comprehensively cover what, and how we make the technical side of Artsy work.
+
+<!-- more -->
+
+# Organizational Structures
+
+In 2016, we made considerable changes to [the Engineering organizational stack](/blog/2016/03/28/artsy-engineering-organization-stack/). This is still an accurate representation of our team structure.
+
+The only structural changes since then has been the down-playing of the web and mobile practices. The transition in the mobile team to React Native has brought the two so close in terms of execution and ideas that it doesn't make sense to create a strong distinction any more. For example, we used to have two subteams in the product team Collector GMV between iOS and Web, now they are merged.
+
+# Artsy Tech Infrastructure
+
+## Front-End + API
+
+In 2015, [dB][db] published our [2015 technology stack][tech2015], in the core Artsy products, though there are a few interesting changes.
+
+What you see today when you go to [artsy.net](https://artsy.net) is a website built with [Ezel.js](http://ezeljs.com), which is a boilerplate for [Backbone](http://backbonejs.org) projects running on [Node](https://nodejs.org) and using [Express](http://expressjs.com) and [Browserify](http://browserify.org). We used to have separate projects for mobile and desktop web, but now they [are merged][force_merge_pr]. It is hosted on [Heroku](http://heroku.com) and uses [Redis](http://redis.io) for caching. Assets, including artwork images, are served from [Amazon S3](http://aws.amazon.com/s3/) via the [CloudFront CDN](http://aws.amazon.com/cloudfront).
+
+What you see today when you open the Artsy iOS app is a mix of Objective-C, Swift and React Native. Objective-C and Swift continue to provide a lot of over-arching cross-View Controller code. While individual representations of Artsy resources tend to be built in React Native. All of our React Native code uses Relay to handle API integration.
+
+Our core-API is a Rails app using [Grape](https://github.com/intridea/grape) which serves JSON. The API [runs on][aws] [AWS OpsWorks](http://aws.amazon.com/opsworks) and retrieves data from several [MongoDB](http://www.mongodb.com) databases hosted with [Compose](https://www.compose.io). It also uses [Apache Solr](http://lucene.apache.org/solr), [Elastic Search](https://www.elastic.co) and [Google Custom Search](https://www.google.com/cse). The API service also heavily relies on [Memcached](http://memcached.org).
+
+[confirm with joey about ^ - I know there are search changes]
```

Comment:
Rather than exhaustive comments I made a PR to your PR here: https://github.com/artsy/artsy.github.io/pull/326

Feel free to take what you will from that.

I'm happy to fill in other sections where needed too. And maybe @wrgoldstein and/or @cavvia could take a crack at the "Data Pipeline" section.